include/swift/AST/Module.h

```diff
@@ -744,7 +748,7 @@ class FileUnit : public DeclContext {
   // Only allow allocation of FileUnits using the allocator in ASTContext
   // or by doing a placement new.
   void *operator new(size_t Bytes, ASTContext &C,
-                     unsigned Alignment = alignof(FileUnit));
+                     unsigned Alignment = alignof(FileUnit&));
```

Comment:
Okay, this actually *does* look correct—"When alignof is applied to a reference type, the result shall be the alignment of the referenced type." (C++11 [expr.alignof]p3)—but if MSVC is accepting one and not the other I'm a little concerned that it's doing the wrong thing. Can you test that it accepts this code?

```c++
struct Foo {
  bool b;
  static_assert(alignof(Foo&) == alignof(bool), "aligning reference rather than Foo");
};
```